test/files/pos/caseclass_private_constructor.scala

```diff
@@ -0,0 +1,47 @@
+case class A private (i: Int)
+object A {
+  def a = A(1).copy(2) // apply and copy are accessible in companion
+}
```

Comment:
Maybe there should be a test for variant of `A` that checks that I can override `apply` in companion to make it public?